packages/astro/src/core/dev/index.ts

```diff
@@ -24,27 +24,30 @@ export default async function dev(config: AstroConfig, options: DevOptions = { l
 	polyfill(globalThis, {
 		exclude: 'window document',
 	});
-	// start the server
+
+	// TODO: remove call once --hostname is baselined
+	const host = getResolvedHostForVite(config);
 	const viteUserConfig = vite.mergeConfig(
 		{
 			mode: 'development',
-			server: {
-				host: config.devOptions.hostname,
-			},
+			server: { host },
 		},
 		config.vite || {}
 	);
 	const viteConfig = await createVite(viteUserConfig, { astroConfig: config, logging: options.logging, mode: 'dev' });
 	const viteServer = await vite.createServer(viteConfig);
 	await viteServer.listen(config.devOptions.port);
+
 	const address = viteServer.httpServer!.address() as AddressInfo;
-	const localAddress = getLocalAddress(address.address, config.devOptions.hostname);
-	// Log to console
+	const localAddress = getLocalAddress(address.address, config);
+	// true - Vite exposes server on default network
+	// string - Vite exposes server on specified network
+	const isNetworkExposed = host === true || typeof host === 'string';
 	const site = config.buildOptions.site ? new URL(config.buildOptions.site) : undefined;
 	info(
 		options.logging,
 		null,
-		msg.devStart({ startupTime: performance.now() - devStart, port: address.port, localAddress, networkAddress: address.address, site, https: !!viteUserConfig.server?.https })
+		msg.devStart({ startupTime: performance.now() - devStart, port: address.port, localAddress, isNetworkExposed, site, https: !!viteUserConfig.server?.https })
```

Comment:
I thought that `--host 1.2.3.4` would work. How does `devStart` know to show the correct `http://1.2.3.4` URL in the console log if we don't share the host string? Is `os.networkInterfaces()` that smart?

Reply:
Yep, `os.networkInterfaces` really does cover us! This is taken directly from Vite's CLI.
For instance, passing my device IP will only expose the server on my network (not localhost or `0.0.0.0`):

<img width="446" alt="Screen Shot 2022-03-11 at 9 00 53 AM" src="https://user-images.githubusercontent.com/51384119/157882305-65028885-797e-4c10-8c95-4d17ee35428e.png">

Reply:
cool! TIL